### PR TITLE
Fixed coding standard issues

### PR DESCRIPTION
**Pull request by sundaramrajendran**
- Enforces calling super inside constructor
- Prohibitis reassigning values to parameters
- Prohibits if-else when returning boolean
- Prohibitis return of internal arrays
- Enforces never changed fields to final
- Prohibits unnecessary comparisions in boolean expressions
- Enforces literals first in comparisons
- Prohibitis non thread safe singletons
- Import statements allow the use of non-fully qualified names
- Use assertSame/assetNotSame in assertions when testing object referrences equality
- Use asserrTrue/assertFalse to assert boolean values
- Avoid unused import statements

### CODE REVIEW

File: client/src/main/java/com/paypal/selion/internal/platform/grid/browsercapabilities/SelendroidCapabilitiesBuilder.java [hunk 1]
```diff
@@ -35,30 +35,31 @@
     @Override
     public DesiredCapabilities getCapabilities(DesiredCapabilities capabilities) {
 
+        DesiredCapabilities tempCapabilities = capabilities;
```
Comment (mach6) [inline client/src/main/java/com/paypal/selion/internal/platform/grid/browsercapabilities/SelendroidCapabilitiesBuilder.java]: Here the _Object reference_  to `capabilities` is copied. So `capabilities` and `tempCapabilities`, refer to the same object.

Thus, this change is not really fixing the `Avoid reassigning parameters such as 'capabilities'` condition.

In this particular case, the approach should be 

``` java
DesiredCapabilties newCapabilities = SelendroidCapabilities.android(); // Note: SelendroidCapabilities.android() returns a new DesiredCapabilties object. So,now that the original `capabilities` _Object reference_ and values are lost.
newCapabilities.merge(capabilities); // I suspect the fact that we don't merge the pre-existing values is a bug we have today. See `IOSDriverCapabilitiesBuilder` -- it calls `merge()`, for example

// the rest of the changes
```

File: client/src/main/java/com/paypal/selion/internal/reports/excelreport/DetailsReport.java [hunk 2]
```diff
@@ -62,17 +62,18 @@ public DetailsReport(String sReportName) {
 
     int fillData(HSSFSheet sheet, int rowNum, HSSFCellStyle style) {
         logger.entering(new Object[] { sheet, rowNum, style });
+        HSSFCellStyle tempStyle = style;
```
Comment (mach6) [inline client/src/main/java/com/paypal/selion/internal/reports/excelreport/DetailsReport.java]: Problematic: _Object reference_ copy

File: client/src/main/java/com/paypal/selion/platform/html/support/HtmlElementUtils.java [hunk 3]
```diff
@@ -187,40 +187,41 @@ public static By resolveByType(String locator) {
      */
     public static By getFindElementType(String locator) {
         logger.entering(locator);
-        Preconditions.checkArgument(StringUtils.isNotBlank(locator), INVALID_LOCATOR_ERR_MSG);
+        String tempLocator = locator;
```
Comment (mach6) [inline client/src/main/java/com/paypal/selion/platform/html/support/HtmlElementUtils.java]: Let's give this a better name (like `seleniumLocator`) and create this object after the `Precondition` where the assignment happens.

File: client/src/main/java/com/paypal/selion/platform/mobile/ios/UIAPicker.java [hunk 4]
```diff
@@ -72,10 +72,11 @@ private WebElement findWheelAtIndex(int index) {
     }
 
     private List<String> asList(String allValues) {
+        String tempAllValues = allValues;
```
Comment (mach6) [inline client/src/main/java/com/paypal/selion/platform/mobile/ios/UIAPicker.java]: Just declare a short-lived variable where the assignment happens, in the `if` loop

File: server/src/main/java/com/paypal/selion/grid/FileExtractor.java [hunk 5]
```diff
@@ -81,6 +81,7 @@ static String getFileNameFromPath(String name) {
         LOGGER.entering(archiveFile);
 
         LOGGER.info("Extracting " + archiveFile);
+        String tempArchiveFile = archiveFile;
```
Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/FileExtractor.java]: Similar feedback. Let's do the assignment to a new variable when/where the assignment happens. Let's keep the 'reads' / conditionals in this method acting on the method parameter `archiveFile`

**Comment by mach6** [comment]
@sundaramrajendran just some minor feedback. otherwise this looks good. thanks for going after the codacy issues!

**Comment by sundaramrajendran** [comment]
Addressed all the review comments

**Comment by mach6** [comment]
updated and pushed to branch https://github.com/paypal/SeLion/tree/sundaramrajendran-raj-coday-fix-errorprone

**Comment by mach6** [comment]
merged to `develop`

**Pull request closed without merging**